### DISCUSSION TITLE
Debian 12 - Installation from source

### DISCUSSION THREAD

**Discussion by rotten777** [agersant/polaris#203] [General]
I've gone through the installation process and created a systemd file for the service but it fails to run as a daemon unless I manually launch the executable from the shell.

Here's the systemd service file (/etc/systemd/system/polaris.service)

```
[Unit]
Description=Polaris Media Server
Wants=basic.target network-online.target nss-lookup.target time-sync.target
After=basic.target network.target network-online.target
Before=sshd.service

[Service]
SyslogIdentifier=polaris
ExecStart=/usr/local/bin/polaris
Restart=always

[Install]
WantedBy=multi-user.target

```

I can't seem to find any docs on setting up the systemd service but I've used this script for other services with success. Any idea if I'm missing something?

The version I'm using is 0.14.2

SOLVED:

```
[Unit]
Description=Polaris
Wants=basic.target network-online.target nss-lookup.target time-sync.target
After=basic.target network.target network-online.target
Before=sshd.service

[Service]
SyslogIdentifier=polaris
Type=forking
ExecStart=/usr/local/bin/polaris
Restart=always

[Install]
WantedBy=multi-user.target

```

**Comment by lnicola** [comment]
This is mine, for an older version of Polaris running on Arch Linux:

```
[Unit]
Description=Polaris
After=network.target

[Service]
User=polaris
Type=notify
ExecStart=/usr/local/bin/polaris -f -w /usr/share/polaris/web -s /usr/share/polaris/swagger
WorkingDirectory=/var/lib/polaris

[Install]
WantedBy=multi-user.target
```

But you're gonna have to expand a little. How does it fail to run?

**Reply by saecki** [reply]
I've got a hunch that the `-f` flag might be the key here.

**Reply by lnicola** [reply]
I think you're right, it should have `Type=forking` if you're not using `-f`.

**Reply by rotten777** [reply]
That fixed it. Thanks for the help!
